include/picongpu/particles/pusher/particlePusherHC.hpp

```diff
@@ -0,0 +1,139 @@
+        const T_FunctorFieldE functorEField,
+        T_Particle & particle,
+        T_Pos & pos,
+        const uint32_t
+    )
+    {
+        float_X const weighting = particle[ weighting_ ];
+        float_X const mass = attribute::getMass( weighting , particle );
+        float_X const charge = attribute::getCharge( weighting , particle );
+
+        using MomType = momentum::type;
+        MomType const mom = particle[ momentum_ ];
+
+        auto bField  = functorBField(pos);
+        auto eField  = functorEField(pos);
+
+        const float_X deltaT = DELTA_T;
+
+
+        Gamma gamma;
+
+        /*
+         * Momentum update
+         * Notation is according to Ripperda's paper
+         */
+        // First half electric field acceleration
+        const sqrt_HC::float3_X mom_minus = precisionCast<sqrt_HC::float_X>( mom + float_X(0.5) * charge * eField * deltaT );
+
+        // Auxiliary quantitites
+        const sqrt_HC::float_X gamma_minus = gamma( mom_minus , mass );
+
+        const sqrt_HC::float3_X tau = precisionCast<sqrt_HC::float_X>( float_X(0.5) * bField * charge * deltaT / mass );
+
+        const sqrt_HC::float_X sigma = pmacc::math::abs2( gamma_minus ) - pmacc::math::abs2( tau );
+
+        const sqrt_HC::float_X u_star = pmacc::math::dot( mom_minus, tau ) / precisionCast<sqrt_HC::float_X>( mass * SPEED_OF_LIGHT );
+
+        const sqrt_HC::float_X gamma_plus = math::sqrt(
+                sqrt_HC::float_X(0.5) * ( sigma + math::sqrt(
+                        pmacc::math::abs2( sigma ) + sqrt_HC::float_X(4.0) * ( pmacc::math::abs2( tau ) + pmacc::math::abs2( u_star ) )
+                    ) )
+            );
+
+        const sqrt_HC::float3_X t_vector = tau / gamma_plus;
+
+        const sqrt_HC::float_X s = sqrt_HC::float_X(1.0) / ( sqrt_HC::float_X(1.0) + pmacc::math::abs2( t_vector ) );
+
+        // Rotation step
+        const sqrt_HC::float3_X mom_plus = s * ( mom_minus
+                + pmacc::math::dot( mom_minus , t_vector ) * t_vector
+                + pmacc::math::cross( mom_minus , t_vector )
+            );
+
+        // Second half electric field acceleration (Note correction mom_minus -> mom_plus here compared to Ripperda)
+        const MomType mom_diff1 = float_X(0.5) * charge * eField * deltaT;
+        const MomType mom_diff2 = precisionCast<float_X>( pmacc::math::cross( mom_plus , t_vector ) );
+        const MomType mom_diff = mom_diff1 + mom_diff2;
+
+        const MomType new_mom = precisionCast<float_X>( mom_plus ) + mom_diff;
+
+        particle[ momentum_ ] = new_mom;
+
+        /*
+         * Position update
+         */
+        Velocity velocity;
+
+        const float3_X vel = velocity( new_mom , mass );
+
+        for( uint32_t d=0 ; d<simDim ; ++d )
+        {
+            pos[d] += ( vel[d] * deltaT ) / cellSize[d];
+        }
+    }
+
+    static pmacc::traits::StringProperty getStringProperties()
+    {
+        pmacc::traits::StringProperty propList( "name", "HC" );
```

Comment:
a verbose name would be more intuitive

Reply:
Agree. I would actually like to have `HigueraCary` instead of `HC` as the pushed name. That would be consistent with other pushers named by the last name (except Axel's).